tfhe/src/core_crypto/algorithms/misc.rs

```diff
@@ -132,6 +132,40 @@ pub fn check_content_respects_mod<Scalar: UnsignedInteger, Input: AsRef<[Scalar]
     }
 }
 
+/// This function converts an unsigned integer to a float value but does so selecting the truncated
+/// value of the input integer, meaning it will not try to round to the closest representable
+/// integer by the floating point type, it will always select the closest representable integer by
+/// the given floating point type that is inferior to the input integer.
+///
+/// This is used to get an approximation of an integer modulus in the float domain that is
+/// guaranteed to not be greater than the integer value.
```

Comment:
Is there a concrete example that could illustrate this ?

Reply:
any integer that would round up when converted to f64 could yield f64 values that overflow when converted back to integers

basically we do the multiplication in the float domain as the gaussian values are f64, but as soon as you use big integers (which is basically always the case for us) then the float has some error and we could end up with a value outside the modulus which is bad 

https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=6dd53516fd4a07c9cc0b92d052fed24f

Reply:
ok